### PR TITLE
Change AbstractBuild to Run

### PR DESCRIPTION
**Pull request by rodrigc**
This is for compatibility with workflow/pipeline plugin.

### CODE REVIEW

**Comment by rodrigc** [comment]
This patch was confirmed by E K to work with the pipeline (formerly workflow) plugin:

https://groups.google.com/d/msg/jenkinsci-users/gwhMuq4KPi8/NKVh_dPQFwAJ

**Comment by dnozay** [comment]
@rodrigc can you please add a test? Looks like there is a repro testcase on the forum. Thank you

**Comment by rodrigc** [comment]
The test involves invoking a build-flow from inside pipeline (formerly workflow) DSL.
Is it OK to add a test-only dependency on the pipeline plugin?

**Comment by dnozay** [comment]
@rodrigc yes, please add a comment inside the added block in the pom. thanks you.

**Comment by rodrigc** [comment]
@dnozay I have added the test which uses pipeline (formerly workflow).  In addition, I fixed some of the tests, which were failing.

File: src/test/groovy/com/cloudbees/plugins/flow/ConcurrencyTest.groovy [hunk 1]
```diff
@@ -97,20 +102,21 @@ class ConcurrencyTest extends DSLTestCase {
...
         def concjob2 = createBlockingJob("concjob2", f1)
 
-        BuildFlow flow = new BuildFlow(Jenkins.instance, getName())
+        BuildFlow flow = new BuildFlow(Jenkins.instance, name.getMethodName())
         flow.concurrentBuild = true;
         flow.dsl = """  build("concjob2")  """
+        flow.onCreatedFromScratch() // need this to updateTransientActions 
 
-        def sfr1 = flow.scheduleBuild2(0)
+        def sfr1 = flow.scheduleBuild2(0, new Cause.UserIdCause())
```
Comment (dnozay) [inline src/test/groovy/com/cloudbees/plugins/flow/ConcurrencyTest.groovy]: hmmmm, well, the first argument is supposed to be the `quietPeriod`, I'm not sure it should be `0` based on the test description.

**Comment by dnozay** [comment]
The only thing blocking this to be merged is the concurrency test that is now bogus and needs to get fixed somehow.

**Comment by rodrigc** [comment]
@dnozay I think the concurrency test which is failing should just be deleted.  See: https://github.com/jenkinsci/git-plugin/commit/4fe33a1fa31459563b7d9e4c4c652b20bde6df4d

**Comment by rodrigc** [comment]
@dnozay can you take a look at this pull request?  I believe that I have addressed all your concerns.
This change will allow people to call a build-flow job from within a pipeline.  It is a way to allow people to slowly migrate to pipeline.

**Comment by dnozay** [comment]
This looks fine to me. Please feel free to take maintainership and perform a new release.

**Comment by dnozay** [comment]
@rodrigc 

Since I am currently looking for someone else to take over as a maintainer; I will not be the one merging this change. Please check <http://jenkins-ci.org/pull-request-greeting>.

Side note: I think introducing new features in this plugin is risky and takes away resources that could be better spent on the [pipeline plugin](https://github.com/jenkinsci/pipeline-plugin). However, please feel free to merge if you are interested in carrying on the flag.
--
Well, since this adds more compatibility to the workflow / pipeline plugin, that change would certainly help the transition. Thank you for removing the flaky test.

**Comment by rodrigc** [comment]
I agree with you that new features should not be added to buildflow-plugin.
People should start transitioning to Pipeline, because that is where the core Jenkins developers
are focusing their efforts.
Bugfixes (which do not add new features) to build-flow plugin are probably OK, since people
have created many build flows in the past few years when workflow/pipeline did not exist.

At least my change will facilitate transition to Pipeline and does not add new features.

**Comment by rodrigc** [comment]
@dnozay If you are OK with me taking maintainership of this plugin, then I will do it.  If you agree, please respond on this thread, and I will point the Jenkins admins to it.
I will consider this plugin as deprecated in favor of Pipeline, but apply minimal bugfixes, since
many people have written build-flows over the years when Pipeline did not exist.

**Comment by dnozay** [comment]
@rodrigc go for it. I have already updated the Readme, wiki, and posted on the ML. Thank you for stepping up. 👍

**Pull request merged**